src/player_activity.cpp

```diff
@@ -318,6 +318,11 @@ void player_activity::finish( player *p )
             p->do_read(&(p->i_at(position)));
             if (type == ACT_NULL) {
                 add_msg(_("You finish reading."));
+
+                // below code added in to support reading from ground/vehicles
+                if ( values[1] != FRM_INV ) {
+                    g->drop(p->inv.size()-1);
```

Comment:
That issue is separate from drago888's change, but yeah, I knew that was possible when I moved the code off invlets with PR #4844.  Using position and trying to workaround situations where position changed was better than the alternative of having a 52 item limit.

I suspect you can intentionally trigger index-off errors now by using multidrop to drop a worn backpack & other stuff when at your encumbrance limit -- if losing the backpack triggers random items to drop, then the indexes will be off and you might drop the wrong things or be out-of-bounds on the inv index.  I tried to be clever in multidrop style situations by dropping higher indexes first, but it's basically a bunch of hacks.  The ultimate fix would be consistent, monotonically increasing per-item ids, but that'd be a huge change and have its own cons.